src/devices/Dhtxx/DhtSensor.cs

```diff
@@ -0,0 +1,307 @@
+        /// </remarks>
+        public bool TryGetTemperatureInFarenheitHumidity(out double temperatureInFarenheit, out double humidity)
+        {
+            var ret = ReadData();
+            temperatureInFarenheit = TemperatureInFarenheit;
+            humidity = Humidity;
+            return ret;
+        }
+
+        /// <summary>
+        /// Create a DHT sensor
+        /// </summary>
+        /// <param name="pin">The pin number (GPIO number)</param>
+        /// <param name="dhtType">The DHT Type, either Dht11 or Dht22</param>
+        public DHTSensor(int pin, DhtType dhtType)
+        {
+            this._pin = pin;
+            this._dhtType = dhtType;
+            _controller.OpenPin(pin);
+        }
+
+        /// <summary>
+        /// Start a reading
+        /// </summary>
+        /// <returns>
+        /// <c>true</c> if read is successfull, otherwise <c>false</c>.
+        /// </returns>
+        public bool ReadData()
+        {
+
+            // Set the max value for waiting micro second
+            // 27 = debug
+            // 99 = release
+            byte waitMS = 99;
+#if DEBUG
+            waitMS = 27;
+#endif
+            _stopwatch.Start();
+            PinValue lststate = PinValue.High;
+            uint counter = 0;
+            byte j = 0, i;
+            for (i = 0; i < 5; i++)
+                _dht11_val[i] = 0;
+
+            // write on the pin
+            _controller.SetPinMode(_pin, PinMode.Output);
+            _controller.Write(_pin, PinValue.Low);
+            //wait 18 milliseconds
+            Wait(18);
+            _controller.Write(_pin, PinValue.High);
+            // Wait about 40 microseconds
+            Wait(0.03);
+            _controller.SetPinMode(_pin, PinMode.Input);
+            for (i = 0; i < MAX_TIME; i++)
+            {
+                counter = 0;
+                while (_controller.Read(_pin) == lststate)
+                {
+                    counter++;
+                    // This wait about 1 micro second
+                    // No other way to do it for such a precision
+                    for (byte wt = 0; wt < waitMS; wt++)
+                        ;
+                    if (counter == MAX_WAIT)
+                        break;
+                }
+                lststate = _controller.Read(_pin);
+                if (counter == MAX_WAIT)
+                    break;
+
+                // top 3 transistions are ignored   
+                if ((i >= 4) && (i % 2 == 0))
+                {
+                    _dht11_val[j / 8] <<= 1;
+                    if (counter > 16)
+                        _dht11_val[j / 8] |= 1;
+                    j++;
+                }
+            }
+            _stopwatch.Stop();
+
+            if ((j >= 40) && (_dht11_val[4] == ((_dht11_val[0] + _dht11_val[1] + _dht11_val[2] + _dht11_val[3]) & 0xFF)))
+            {
+                if ((_dht11_val[0] == 0) && (_dht11_val[2] == 0))
```

Comment:
`LastReadSuccessful = (_dht11_val[0] != 0) || (_dht11_val[2] != 0);`

Reply:
you can probably also add the previous read as a one liner but that would probably be slightly confusing :-)